### PR TITLE
Move from a "github config" to a "munge object"

### PR DESCRIPTION
**Pull request by eparis**
- Label removals in one munger will be 'seen' in the next
- Locally cache github results and use conditional gets
- Significant improvements to submit-queue web interface (won't crash your browser after a while)
- More debug and visibility information for users concerning what the bot is up to.

### CODE REVIEW

**Comment by eparis** [comment]
@lavalamp ask to do reviews and you shall receive   :)

**Comment by eparis** [comment]
@lavalamp @brendandburns huge improvements, including being able to run the bot every 10 minutes instead of every 30 and the web site won't crash your browser after a while any more.

Can we get a merge/build/run and if it fails, just roll back. The bot won't do anything new/crazier than it did, but we'll have a lot more idea what it's up to...

**Comment by eparis** [comment]
http://plnkr.co/edit/M9xY3oM8e9VChnPXJsFt?p=preview is a pretty up2date example of the UI features.

File: mungegithub/github/github.go [hunk 1]
```diff
@@ -38,28 +38,90 @@ import (
...
+// By default github responds to PR requests with:
+//    Cache-Control:[private, max-age=60, s-maxage=60]
+// Which means the httpcache would not consider anything stale for 60 seconds.
+// However, when we re-check 'PR.mergeable' we need to skip the cache.
+// I considered checking the req.URL.Path and only setting max-age=0 when
+// getting a PR or getting the CombinedStatus, as these are the times we need
+// a super fresh copy. But since all of the other calls are only going to be made
+// once per poll loop the 60 second github freshness doesn't matter. So I can't
+// think of a reason not to just keep this simple and always set max-age=0 on
+// every request.
```
Comment (eparis) [inline mungegithub/github/github.go]: discussion why i believe we can start using gregjones/httpcache again (to get the huge speedup)

**Comment by lavalamp** [comment]
Wow I'm not sure how this evaded my filter the first time around, but I didn't see it until just now. Sorry!

@brendandburns can you take this or delegate? I'm going to be OOO tomorrow.

**Pull request merged**